hack/build-and-push.sh

```diff
@@ -58,15 +59,12 @@ if [[ "${BASE_REF}" != "master" ]]
 then
     # Since we know this is built from a tag or release branch, we can set the VERSION_TAG
     VERSION_TAG="${BASE_REF}"
+
     # We want the binary version to show up correctly too.
     BINARY_VERSION="${BASE_REF}"
-    # Use some bash magic to check if the semver does not end with -sometext, that
-    # would indicate a prerelease version. If this is not a prerelease, then we want to set
-    # the `latest` tag too.
-    if [[ ! "${BASE_REF}" =~ -(.+)$ ]];
-    then
-    LATEST=true
-    fi
+
+    # If we're on a semver baseref, then we don't want to tag the image with :latest
+    LATEST=false
```

Comment:
Does this mean we won't tag latest as part of semver release like v0.5.0?

Reply:
Yes, this means that `:latest` is the latest dev build rather than the latest stable. Maybe this is too far outside the usual contract? I think that using `:latest` for things you want to be stable is an antipattern anyway.

Reply:
I usually think of `:latest` representing the latest build on the master/main branch. If I'm interpreting this change correctly, we'd actually end up pushing a new image anytime a commit merged to a different branch like release-0.4. I agree that relying on a `:latest` image is a bad idea in general, but I'd generally expect updates to that tag to be relatively sequential in nature, so a latest referencing a master commit getting replaced by one that references a release-0.4 commit would be surprising.

I know it's also fairly common for `:latest` to represent the latest release, not the latest commit, but since these images are already limited to the staging repo, it's probably fine to just track the latest commit.

Reply:
Ah, you're right, I have a logic error, I need to check if we're on master, and set LATEST to `true` in that case only, not assume it's true and set it to false if semver. I'll update.